mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1206,74 +1323,92 @@ static LogicalResult populateValidationLogic(
   block->push_back(inputMemRefCastOp);
   block->push_back(outputMemRefCastOp);
 
-  float f1 = 1.0;
-  float f0 = 0.0;
-  auto getOneConstOp = [&]() {
-    if (dataType == builder.getIntegerType(16)) {
-      const ushort one = float_to_bfloat16(f1);
-      return builder.create<ConstantOp>(builder.getUnknownLoc(), dataType,
-                                        builder.getI16IntegerAttr(one));
-    } else {
-      return builder.create<ConstantOp>(builder.getUnknownLoc(), dataType,
-                                        builder.getFloatAttr(dataType, f1));
-    }
-  };
-  auto getZeroConstOp = [&]() {
-    if (dataType == builder.getIntegerType(16)) {
-      const ushort zero = float_to_bfloat16(f0);
-      return builder.create<ConstantOp>(builder.getUnknownLoc(), dataType,
-                                        builder.getI16IntegerAttr(zero));
-    } else {
-      return builder.create<ConstantOp>(builder.getUnknownLoc(), dataType,
-                                        builder.getFloatAttr(dataType, f0));
-    }
-  };
+  auto int16Type = builder.getIntegerType(16);
```

Comment:
the logic is duplicated. please try factor out a common function.

Reply:
I've factored the code out in a function "configRandomTestData()"